PWGHF/TableProducer/candidateSelectorXicToPKPi.cxx

```diff
@@ -239,15 +290,45 @@ struct HfCandidateSelectorXicToPKPi {
 
       if (pidXicToPKPi == 0 && pidXicToPiKP == 0) {
         hfSelXicToPKPiCandidate(statusXicToPKPi, statusXicToPiKP);
+        if (applyMl) {
+          hfMlXicToPKPiCandidate(outputMlXicToPKPi, outputMlXicToPiKP);
+        }
         continue;
       }
-
       if ((pidXicToPKPi == -1 || pidXicToPKPi == 1) && topolXicToPKPi) {
         statusXicToPKPi = 1; // identified as Xic->pKpi
       }
       if ((pidXicToPiKP == -1 || pidXicToPiKP == 1) && topolXicToPiKP) {
         statusXicToPiKP = 1; // identified as Xic->piKp
       }
+      SETBIT(statusXicToPKPi, aod::SelectionStep::RecoPID);
+      SETBIT(statusXicToPiKP, aod::SelectionStep::RecoPID);
+
+      if (applyMl) {
+        // ML selections
+        bool isSelectedMlXicToPKPi = false;
+        bool isSelectedMlXicToPiKP = false;
+
+        if (statusXicToPKPi > 0) {
+          std::vector<float> inputFeaturesXicToPKPi = hfMlResponse.getInputFeatures(candidate, trackPos1, trackNeg, trackPos2);
+          isSelectedMlXicToPKPi = hfMlResponse.isSelectedMl(inputFeaturesXicToPKPi, ptCand, outputMlXicToPKPi);
+        }
+        if (statusXicToPiKP > 0) {
+          std::vector<float> inputFeaturesXicToPiKP = hfMlResponse.getInputFeatures(candidate, trackPos1, trackNeg, trackPos2);
+          isSelectedMlXicToPiKP = hfMlResponse.isSelectedMl(inputFeaturesXicToPiKP, ptCand, outputMlXicToPiKP);
+        }
+
+        if (!isSelectedMlXicToPKPi) {
+          statusXicToPKPi = 0;
+        }
+        if (!isSelectedMlXicToPiKP) {
+          statusXicToPiKP = 0;
```

Comment:
This defies the purpose of setting different bits for the different selection "categories" since you overwrite them here.